src/v/raft/tests/raft_fixture.cc

```diff
@@ -85,13 +85,29 @@ ss::future<iobuf> channel::exchange(msg_type type, iobuf request) {
 }
 bool channel::is_valid() const { return _node && _node->raft() != nullptr; }
 
-ss::lw_shared_ptr<consensus> channel::raft() {
+raft::service<fixture_group_manager, fixture_shard_manager>&
+channel::get_service() {
     if (!_node || _node->raft() == nullptr) {
         throw std::runtime_error("no raft group");
     }
-    return _node->raft();
+    return _node->get_service();
 }
 
+namespace {
+struct test_ctx : rpc::streaming_context {
+    test_ctx()
+      : _sem(std::numeric_limits<size_t>::max(), "test_ctx_sem") {}
+    ss::future<ssx::semaphore_units> reserve_memory(size_t) final {
+        return ss::get_units(_sem, 1);
```

Comment:
nit: Why not just return `ssx::semaphore_units()` instead, so we don't need `_sem`?

Reply:
i didn't notice default constructor, ty